### PR TITLE
Feature/homepage styles/noticket

### PR DESCRIPTION
**Pull request by alexisBimatovs**
Updates to support homepage
- added angle treatment
- expanded button to include not just core

### CODE REVIEW

**Comment by alexisBimatovs** [comment]
to note on this slant in the future if we identify that we want to used in contained elements it will need a refactor, however for the current full page treatment it seems to work :)

**Comment by schweitzdesign** [comment]
The built gems were comitted here, I assume this was incidental since this is unnecessary. These files should not be included.

screen shot 2017-08-07 at 5 00 12 pm

**Comment by schweitzdesign** [comment]
Neither version.rb nor Gemfile.lock are updated here and that's necessary for the version bump.

File: app/assets/stylesheets/forever_style_guide/modules/_button.scss [hunk 1]
```diff
@@ -57,7 +57,7 @@
   }
 
   // creates a button style for each color in core color dictionary
-  @each $id, $color in $core_colors {
+  @each $id, $color in $colors {
```
Comment (schweitzdesign) [inline app/assets/stylesheets/forever_style_guide/modules/_button.scss]: This produces a bunch of errors in the rails console akin to the following:

```
WARNING: Unknown `live17-purple-light-dark` in $colors.
Backtrace:
	../forever-style-guide/app/assets/stylesheets/forever_style_guide/utils/_color-utils.scss:40, in function `get_color`
	../forever-style-guide/app/assets/stylesheets/forever_style_guide/utils/_color-utils.scss:71, in function `color`
	../forever-style-guide/app/assets/stylesheets/forever_style_guide/modules/_button.scss:10, in mixin `btn-color`
	../forever-style-guide/app/assets/stylesheets/forever_style_guide/modules/_button.scss:65
```

File: app/assets/stylesheets/forever_style_guide/globals/_effects.scss [hunk 2]
```diff
@@ -161,3 +161,37 @@ $color-card-bg: color('white');
...
+      width: 0;
+      height: 0;
+      border-right: 100vw solid transparent;
+      border-bottom: 60px solid color('#{$id}');
+      top: -60px;
+    }
+  }
+
+  .slant-left-#{$id} {
+  width: 100%;
```
Comment (schweitzdesign) [inline app/assets/stylesheets/forever_style_guide/globals/_effects.scss]: indentation

File: app/assets/stylesheets/forever_style_guide/globals/_effects.scss [hunk 3]
```diff
@@ -161,3 +161,37 @@ $color-card-bg: color('white');
...
+  position: relative;
+  
+    &::after {
+      content: "";
+      position: absolute;
+      width: 0;
+      height: 0;
+      border-left: 100vw solid transparent;
+      border-bottom: 60px solid color('#{$id}');
```
Comment (schweitzdesign) [inline app/assets/stylesheets/forever_style_guide/globals/_effects.scss]: I would suggest replacing all of thes 60's w/ a variable or, probably better, making this into a mixin that accepts a value to define how steep you want the effect. Alternatively, you could provide a standard rhythm for decreasing the steepness at given breakpoints.

In the current implementation it seems way too steep at any non-desktop breakpoint.

screen shot 2017-08-07 at 5 36 18 pm

File: app/assets/stylesheets/forever_style_guide/globals/_effects.scss [hunk 4]
```diff
@@ -161,3 +161,37 @@ $color-card-bg: color('white');
...
+  .slant-left-#{$id} {
+  width: 100%;
+  position: relative;
+  
+    &::after {
+      content: "";
+      position: absolute;
+      width: 0;
+      height: 0;
+      border-left: 100vw solid transparent;
```
Comment (schweitzdesign) [inline app/assets/stylesheets/forever_style_guide/globals/_effects.scss]: We don't use vw units anywhere else and they are not well supported enough http://caniuse.com/#search=vw

File: app/assets/stylesheets/forever_style_guide/globals/_effects.scss [hunk 5]
```diff
@@ -161,3 +161,37 @@ $color-card-bg: color('white');
     }
   }
 }
+
+// can be added to sections to create slant effect
+@each $id, $color in $colors {
+
+  .slant-right-#{$id} {
+  width: 100%;
```
Comment (schweitzdesign) [inline app/assets/stylesheets/forever_style_guide/globals/_effects.scss]: indentation

File: app/assets/stylesheets/forever_style_guide/globals/_effects.scss [hunk 6]
```diff
@@ -161,3 +161,37 @@ $color-card-bg: color('white');
     }
   }
 }
+
+// can be added to sections to create slant effect
```
Comment (schweitzdesign) [inline app/assets/stylesheets/forever_style_guide/globals/_effects.scss]: This comment should clarify that the affect is applied to the TOP of the element to which you assign this class. We should also include a visual example of this in the style guide dummy app.

File: app/assets/stylesheets/forever_style_guide/globals/_effects.scss [hunk 7]
```diff
@@ -161,3 +161,37 @@ $color-card-bg: color('white');
...
+      top: -60px;
+    }
+  }
+
+  .slant-left-#{$id} {
+  width: 100%;
+  position: relative;
+  
+    &::after {
+      content: "";
```
Comment (schweitzdesign) [inline app/assets/stylesheets/forever_style_guide/globals/_effects.scss]: This whole effect can also be DRYed up a lot to reduce the redundant code between left and rigth variations.

File: app/assets/stylesheets/forever_style_guide/globals/_effects.scss (hunk 4 above)
Comment (alexisBimatovs) [inline app/assets/stylesheets/forever_style_guide/globals/_effects.scss]: weird seems like supporting browsers differ from these links: https://www.w3schools.com/cssref/css_units.asp

File: app/assets/stylesheets/forever_style_guide/globals/_effects.scss (hunk 6 above)
Comment (alexisBimatovs) [inline app/assets/stylesheets/forever_style_guide/globals/_effects.scss]: to note I had tried to add a working example in the guide quickly in pervious implementation but position absolute was giving me trouble in the existing layout, with the refactor this should not be an issue but in an effort to focus on other related homepage release items I have logged as a to do to circle back to ASAP: https://www.pivotaltracker.com/story/show/150215444

**Comment by alexisBimatovs** [comment]
@schweitzdesign I sat with Brad and ran through this branch and the feedback was as long as all my commits are grouped and on top, which they are, and the changed files look good, and there are no merge conflicts than I should not try and resolve the additional history beneath my changes in here, could have been from the way I rebased, sorry for any confusion let me know if you are good to review, thanks

File: app/assets/stylesheets/forever_style_guide/globals/_effects.scss [hunk 8]
```diff
@@ -183,11 +179,10 @@ $slant-angle-right: 182deg;
...
   content: "";
+  display: block;
   position: relative;
   top: $slant-position-top;
   height: $slant-height;
-  display: block;
-  margin-top: -1px;
   background-color: inherit;
 
   @media (min-width: $screen-xxl) {
```
Comment (alexisBimatovs) [inline app/assets/stylesheets/forever_style_guide/globals/_effects.scss]: empty media query?

**Pull request merged**